Automattic/woocommerce-payments
# added check to prevent redirecting the customer automatically before …

Fixes #5140

#### Changes proposed in this Pull Request
This PR changes the order in which the iframes and checks are executed

Before:
- Check if they are logged in to WooPay
- If so then redirect them to WooPay
- If not logged in to WooPay check if the user exists/is logged in to the merchant site
- Show the OTP modal 

Now:
- Check if the user exists/is logged in to the merchant site
- Check if they are logged in to WooPay
- If so then redirect them to WooPay
- If not logged in to WooPay show the OTP modal 

#### Testing instructions
- As a shopper while logged out from WooPay and the merchant site
- Add a product to the cart
- Go to checkout 
- Type a WooPay account email
- Wait until the OTP modal shows up
- Make sure the spinner was displayed until the modal showed up
- Type the OTP
- Make sure you get redirected to WooPay
- Make sure no errors show up
- Click the browser back button
- Make sure you don't get redirected back to WooPay and the OTP modal does not show up
- Delete a character and add it back to trigger the input listener
- Make sure the input spinner keep showing until the redirect modal showed up
- Make sure you get redirected to WooPay again
- Click the browser back button again
- Reload the checkout page
- While the spinner is being displayed trigger the input listener again
- Make sure you still gets redirected to WooPay

-------------------


**Post merge**
